### ISSUE TITLE
Does it need to execute the transformer model to decompress files?

### ISSUE THREAD

**Issue by suntong30** [mynotwo/A-Fast-Transformer-based-General-Purpose-LosslessCompressor#1]
Hello, I am interested in this work, but I am not familiar with DNN-based compressors. I know that we need the transformer model to compress the data. 

In standard compressors, such as zlib and 7zip, utilize deflate and inflate algorithms to compress and decompress. So, in the DNN-based compressors, do we also need the model to decompress the compressed data?

**Comment by mynotwo** [comment]
Yes, you'll need the model to predict the exact same probability when decompress.

**Comment by suntong30** [comment]
Hello, I have the following question, is the model only trained on the input data? For example, if the input file is "dickens", the model reads the "dickens" and trains on it to minimize the loss function. It doesn't need pre-trained. Am I right?

Best regards!

**Comment by mynotwo** [comment]
Yes! But if you want higher compression ratio, you can pre-train on dickens for several epochs, and then do the compression. The compression ratio report in my paper does not include pre-training.

**Comment by BitCalSaul** [comment]
@mynotwo hi thanks for your work. im wondering the way you handle with the unpredictable data. even if the model is quite big, there is still some data that your model may handle not well. so to keep this compressor lossless, how do you manage this part of data? thx
